src/main/java/org/broadinstitute/hellbender/engine/ReadsContext.java

```diff
@@ -99,4 +100,35 @@ public SimpleInterval getInterval() {
                 dataSource.query(interval) :
                 new ReadFilteringIterator(dataSource.query(interval), readFilter);
     }
+
+    /**
+     * Create a map of reads overlapping {@code interval} to their mates by looking for all possible mates within some
+     * maximum fragment size.  This is not guaranteed to find all mates, in particular near structural variant breakpoints
+     * where mates may align far away.
+     *
+     * The algorithm is:
+     * 1) make two maps of read name --> read for reads overlapping {@code interval}, one for first-of-pair reads and one
+     *    for second-of-pair reads.
+     * 2) For all reads in an expanded interval padded by {@code fragmentSize} on both sides look for a read of the same
+     *    that is second-of-pair if this read is first-of-pair or vice-versa.  If such a read is found then this is that read's mate.
+     *
+     * @param fragmentSize the maximum distance on either side of {@code interval} to look for mates.
+     * @return a map of reads ot their mates for all reads for which a mate could be found.
+     */
+    public Map<GATKRead, GATKRead> getReadToMateMap(final int fragmentSize) {
```

Comment:
This might be more at home in `ReadUtils` rather than here, as it's a pretty specific method. You could add a generic `query(interval)` method to `ReadsContext` to enable this method to move to `ReadUtils`.

Reply:
Done